test/jaxp/javax/xml/jaxp/unittest/xpath/XPathExpParentTest.java

```diff
@@ -0,0 +1,159 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package xpath;
+
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathFactory;
+import javax.xml.xpath.XPathNodes;
+
+/*
+ * @test
+ * @bug     8292990
+ * @summary Tests the XPath parent axis
+ * @library /javax/xml/jaxp/unittest
+ *
+ * @run testng xpath.XPathExpParentTest
+ */
+public class XPathExpParentTest extends XPathTestBase {
+    private static final Document doc = getDocument();
+
+    /*
+     * DataProvider for XPath expressions which should provide one and only
+     * one parent node.
+     * Data columns:
+     *  see parameters of the test "testOneParentNodeExp"
+     */
+    @DataProvider(name = "oneParentNode")
+    public Object[][] getOneParentNodeExp() {
+        return new Object[][]{
+                {"//Customer/parent::*", "//Customers"},
+                {"//Customer[1]/text()/parent::*", "//Customer"},
+                {"//Customer[1]/@id/parent::*", "//Customer"},
+                {"//Customer[1]/namespace::*/parent::*", "//Customers"},
+                {"/Customers/comment()/parent::*", "//Customers"},
+                {"//Customer[1]/..", "//Customers"},
+                {"//Customer[1]/text()/..", "//Customer"},
+                {"//Customer[1]/@id/..", "//Customer"},
+                {"//Customer[1]/namespace::*/..", "//Customers"},
+                {"/Customers/comment()/..", "//Customers"},
+                {"//*[parent::Customers][1]", "//Customer"},
+                {"//*[parent::Customers and @id='x1']", "//Customer"},
+
+                // document root
+                {"/Customers/parent::node()", "/"},
+                {"/Customers/..", "/"},
+        };
+    }
+
+    /*
+     * DataProvider for XPath expressions which should provide no parent node.
+     * Data columns:
+     *  see parameters of the test "testZeroParentNodeExp"
+     */
+    @DataProvider(name = "noParentNode")
+    public Object[][] getZeroParentNodeExp() {
+        return new Object[][]{
+                // parent::* never return document root
+                {"/Customers/parent::*"},
+
+                // parent of document root
+                {"/.."},
```

Comment:
The nature of these two statements seems to be quite different. It might be worth to explain a bit more, a reference to the spec or note from the Slack discussion, sth. that might help the reader of these tests.